lib/heartcheck/webservice/http_client.rb

```diff
@@ -3,17 +3,29 @@ module Heartcheck
   module Webservice
     # Http Client
     class HttpClient
-      attr_reader :uri, :http
+      attr_reader :uri, :http, :proxy, :headers, :request
 
-      def initialize(url, ignore_ssl_cert)
+      def initialize(url, proxy, ignore_ssl_cert, headers)
         @uri = URI url
-        @http =  Net::HTTP.new(@uri.host, @uri.port)
+        @headers = headers || {}
+
+        if proxy
+          @proxy = URI proxy
+          @http = Net::HTTP.new(@uri.host, @uri.port, @proxy.host, @proxy.port)
+        else
+          @http =  Net::HTTP.new(@uri.host, @uri.port)
+        end
```

Comment:
Done. `present?` is not available in ruby, only in Rails so I'm using a regular nil check instead.